contributors/devel/sig-scheduling/scheduling_code_hierarchy_overview.md

```diff
@@ -0,0 +1,280 @@
+for each plugin’s New function. It should be noted that the Registry is simply
+a map of plugin names to their factory functions. For the default scheduler,
+this step does nothing (because our main function in `cmd/kube-scheduler/scheduler.go`
+passes nothing to `NewSchedulerCommand()`).
+This means the default set of plugins is initialized in `scheduler.New()`.
+
+Given the initialization is performed outside the scheduling framework,
+different consumers of the framework can initialize the environment differently
+to cover their needs. For example, a simulator can inject its own object
+through informers. Or custom plugins may be provided instead of the default ones.
+Known consumers of the scheduling framework:
+- [cluster-autoscaler](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/simulator/scheduler_based_predicates_checker.go#L48-L79)
+- [cluster-capacity](https://github.com/kubernetes-sigs/cluster-capacity/blob/8e9c2dcf3644cb5f73fca3d35d4e22899c265ad5/pkg/framework/simulator.go#L370-L383)
+
+### Assembling the scheduler
+
+The code is located under `pkg/scheduler`.
+This is where implementation of the default scheduler lives.
+Various elements of the scheduler are initialized and put together here:
+- Default scheduling options such as node percentage, initial and maximum backoff, profiles
+- Scheduler cache and queues
+- Scheduling profiles instantiated to tailor a framework for each profile
+  to better suit pod placement (each profile defines a set of plugins to use)
+- Handler functions for getting the next pod for scheduling (`NextPod`) and error handling (`Error`)
+
+The following steps are taken during the process of creating a scheduler instance:
+- Scheduler [cache is initialized](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L206)
+- Both in-tree and out-of-tree registries with plugins are [merged together](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L208-L211)
+- Metrics are [registered](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L232)
+- [Configurator](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L215-L230)
+  building a scheduler instances (wiring the cache, plugin registry,
+  scheduling algorithm and other elements together)
+- Event handlers [are registered](https://github.com/kubernetes/kubernetes/blob/a651804427dd9a15bb91e1c4fb7a79994e4817a2/pkg/scheduler/scheduler.go#L273)
+  to allow the scheduler to react on changes in PVs,
+  PVCs, services and other objects relevant for scheduling (eventually,
+  each plugin will define a set of events on which it reacts,
+  see [kubernetes/kubernetes#94009](https://github.com/kubernetes/kubernetes/issues/94009)
```

Comment:
I have a new issue compiling completed and on-going items: https://github.com/kubernetes/kubernetes/issues/100347